### PR TITLE
Fix lint

### PR DESCRIPTION
**Pull request by kokeshiM0chi**
# Overview
Change the cargo clippy check range from `--all-targets --all-features` to `--all`

This pr changes two:
- `cargo clippy` has the same detection range as `cargo fmt`. Set to `cargo clippy --all -- -D warnings`.
- Fixed the code that caused an error with `cargo clippy --all -- -D warnings`

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -14,7 +14,11 @@ jobs:
         with:
           submodules: recursive
       - uses: ./.github/actions/cache
-      - run: rustup component add rustfmt
+      - run: |
+          rustup toolchain add 1.69.0
+          rustup override set 1.69.0
+          rustup component add clippy --toolchain 1.69.0-x86_64-unknown-linux-gnu
+          rustup component add rustfmt --toolchain 1.69.0-x86_64-unknown-linux-gnu
```
Comment (170210) [inline .github/workflows/lint.yaml]:
```suggestion
      - name: Install rust toolchain
        uses: actions-rs/toolchain@v1
        with:
            toolchain: 1.69.0
            override: true
            components: rustfmt, clippy
```

**Review by 170210 — COMMENTED** [review]
Is it still need https://github.com/Finschia/finschia-wasm/pull/14/commits/909044819dc6b6824d880a3805901a2822ec4943 and https://github.com/Finschia/finschia-wasm/pull/14/commits/ad7fcb36e17cc18ed9581ab3bac0c7c64097ff9c. If we only modify rust version, could it work?

**Comment by kokeshiM0chi** [comment]
@170210 
>Is it still need https://github.com/Finschia/finschia-wasm/commit/909044819dc6b6824d880a3805901a2822ec4943 and https://github.com/Finschia/finschia-wasm/commit/ad7fcb36e17cc18ed9581ab3bac0c7c64097ff9c. If we only modify rust version, could it work?

Yes.
The former vector's [first()](https://doc.rust-lang.org/std/primitive.slice.html#method.first) seems to be a standard function introduced in [1.56.0](https://github.com/rust-lang/rust/blob/1.56.0/library/core/src/slice/mod.rs#L142-L144). The latter modification to remove the extra clone() is also required in 1.69.0.
I also confirmed the operation.

**Review by 170210 — APPROVED** [review]

**Comment by kokeshiM0chi** [comment]
This command causes an error. I have issued an issue to the dependency finschia/cosmwasm that is causing the error.
```
cargo clippy --workspace --all-targets --all-features -- -D warnings
```

https://github.com/Finschia/cosmwasm/issues/346

**Pull request merged**